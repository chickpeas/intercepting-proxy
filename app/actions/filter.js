// @flow

export const ENABLE_FILTER = 'ENABLE_FILTER';
export const ENABLE_MIME_FILTER = 'ENABLE_MIME_FILTER';
export const ENABLE_STATUS_FILTER = 'ENABLE_STATUS_FILTER';
export const FORWARD_REQUEST = 'FORWARD_REQUEST';

export function changeInterceptFilter(value: boolean) {
  return {
    type: ENABLE_FILTER,
    payload: {
      value
    }
  };
}

export function changeMimeFilter(value: Object) {
  return {
    type: ENABLE_MIME_FILTER,
    payload: {
      value
    }
  };
}

export function changeStatusFilter(value: Object) {
  return {
    type: ENABLE_STATUS_FILTER,
    payload: {
      value
    }
  };
}
