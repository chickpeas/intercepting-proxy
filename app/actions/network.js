// @flow
import { createId } from './../utils/index';

export const ADD_REQUEST = 'ADD_REQUEST';
export const ADD_PENDING_REQUEST = 'ADD_PENDING_REQUEST';
export const REMOVE_PENDING_REQUEST = 'REMOVE_PENDING_REQUEST';
export const ADD_RESPONSE = 'ADD_RESPONSE';
export const ENABLE_FILTER = 'ENABLE_FILTER';

export function addRequest(id, req) {
  return {
    type: ADD_REQUEST,
    payload: {
      id,
      requestId: createId(),
      method: req.method,
      url: req.url,
      headers: req.headers
    }
  };
}
export function addPendingRequest(id, req) {
  return (dispatch) => {
    dispatch(addRequest(id, req));
    dispatch({
      type: ADD_PENDING_REQUEST,
      payload: {
        id
      }
    });
  };
}

export function forwardRequest() {
  return (dispatch) => {
    dispatch({
      type: REMOVE_PENDING_REQUEST
    });
  };
}

export function addResponse(id, res) {
  return {
    type: ADD_RESPONSE,
    payload: {
      id,
      responseId: createId(),
      statusCode: res.statusCode
    }
  };
}
