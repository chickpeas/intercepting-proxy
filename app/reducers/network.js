// @flow
import { ADD_REQUEST, ADD_RESPONSE } from '../actions/network';

export type networkStateType = {
  network: Object<{
    byId: Array,
    byHash: {
      requestId: string,
      responseId: string
    }
  }>
};

type actionType = {
  type: string,
  payload?: Object
};

const initialState = {
  byId: [],
  byHash: {}
};

export default function network(state: Object = initialState, action: actionType) {
  switch (action.type) {
    case ADD_REQUEST:
      return {
        // update byId only when we have the response
        ...state,
        byId: [
          ...state.byId,
          action.payload.id
        ],
        byHash: {
          ...state.byHash,
          [action.payload.id]: {
            ...state[action.payload.id],
            requestId: action.payload.requestId
          }
        }
      };
    case ADD_RESPONSE:
      return {
        // byId: [
        //   ...state.byId,
        //   action.payload.id
        // ],
        ...state,
        byHash: {
          ...state.byHash,
          [action.payload.id]: {
            ...state.byHash[action.payload.id],
            responseId: action.payload.responseId
          }
        }
      };
    default:
      return state;
  }
}
