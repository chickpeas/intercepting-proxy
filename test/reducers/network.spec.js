import network from '../../app/reducers/network';
import { ADD_REQUEST, ADD_RESPONSE } from '../../app/actions/network';

describe('reducers', () => {
  describe('network', () => {
    it('should handle initial state', () => {
      expect(network(undefined, {})).toMatchSnapshot();
    });

    it('should handle ADD_RESPONSE', () => {
      expect(network({ byId: [], byHash: {} }, { type: ADD_RESPONSE, payload: {} })).toMatchSnapshot();
    });

    it('should handle ADD_REQUEST', () => {
      expect(network({ byId: [], byHash: {} }, { type: ADD_REQUEST, payload: {} })).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(network({ byId: [], byHash: {} }, { type: 'unknown' })).toMatchSnapshot();
    });
  });
});
