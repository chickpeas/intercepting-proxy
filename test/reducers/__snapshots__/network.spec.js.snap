// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers network should handle ADD_REQUEST 1`] = `
Object {
  "byHash": Object {
    "undefined": Object {
      "requestId": undefined,
    },
  },
  "byId": Array [
    undefined,
  ],
}
`;

exports[`reducers network should handle ADD_RESPONSE 1`] = `
Object {
  "byHash": Object {
    "undefined": Object {
      "responseId": undefined,
    },
  },
  "byId": Array [],
}
`;

exports[`reducers network should handle initial state 1`] = `
Object {
  "byHash": Object {},
  "byId": Array [],
}
`;

exports[`reducers network should handle unknown action type 1`] = `
Object {
  "byHash": Object {},
  "byId": Array [],
}
`;
